fontbe/src/features.rs

```diff
@@ -111,6 +113,190 @@ impl<'a> FeaVariationInfo<'a> {
     }
 }
 
+struct FeatureWriter<'a> {
```

Comment:
wouldn't it be better to have distinct feature writers for kern on the one hand and mark/mkmk on the other? As we implement the full logic of the python equivalent this single struct may get become too complex..

Reply:
The way the API currently works, from rust, is that you need to provide a single type that gets the callback; but there's no reason we couldn't split it up on the fontc side.

fwiw this patch was really written as a proof of concept, just to see if the API would work. Happy to see this improved upon going forward :)

Reply:
I'm hoping that some of the work can be done separately so the implementation here is basically go get the kerning (or w/e) we computed earlier and return it.

Reply:
And a big +1 to refinement over time, hopefully in small increments now we've got fea-rs in the same repo so we can iterate on fontbe and fea-rs in a single PR